drgn/helpers/linux/runq.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2025, Oracle and/or its affiliates.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+Run Queues
+----------
+The ``drgn.helpers.linux.runq`` module provides helpers for working with the
+Linux run queues.
+"""
+from drgn import Object, Program
+from drgn.helpers.linux.percpu import per_cpu
+from drgn.helpers.linux.sched import task_cpu
+
+
+def runq_clock(prog: Program, cpu: int) -> int:
+    """
+    Get clock of cpu runqueue ``struct rq``
+
+    :param prog: drgn program
+    :param cpu: cpu index
+    :returns: cpu runqueue clock in ns granularity
+    """
+    rq = per_cpu(prog["runqueues"], cpu)
+    return rq.clock.value_()
```

Comment:
This helper is fairly simple, which I wouldn't mind if there was an equivalent function in the kernel, but that doesn't seem to be the case.

What if instead we add `cpu_rq()` and `task_rq()` helpers based on the existing kernel macros: https://github.com/torvalds/linux/blob/c7de79e662b8681f54196c107281f1e63c26a3db/kernel/sched/sched.h#L1346-L1348

Then, instead of:

```python3
rq_clock = runq_clock(prog, task_cpu(task))
```

You'd do:

```python3
rq_clock = task_rq(task).clock.value_()
```

Which is still an improvement over open-coding the whole thing.